### PR TITLE
Match gnome product names like gedit

### PR DESCRIPTION
**Pull request by pedrocr**
Use the X-GNOME-Bugzilla-Product property of desktop files to be able to match against "gedit" and not just "text editor".

### CODE REVIEW

File: src/widgets/launcher.rs [hunk 1]
```diff
@@ -239,6 +241,16 @@ impl Widget for Launcher {
                     .map(|(x, y)| (x.unwrap(), y.clone()))
                     .collect::<Vec<(i64, Desktop)>>();
 
+                for desktop in self.options.iter() {
```
Comment (pedrocr) [inline src/widgets/launcher.rs:245]: There's probably a better way to do this in a single loop together with the name matches or in a more idiomatic way but it wasn't immediately obvious to me.

**Comment by kennylevinsen** [comment]
Just a note: It seems like the use of X-GNOME-Bugzilla-Product is somewhat incosistent.

The .desktop file for Videos (Totem) has it, for example, but Document Viewer (Evince) does not. Both are from flathub (flatpak), and I believe they are maintained by the GNOME team themselves.

The only field that consistently contain the GNOME *project* names would seem to be Keywords, which would certainly complicate the search/display logic a bit.

**Comment by pedrocr** [comment]
Matching on keywords may fix more cases but that's the one we discussed could also backfire. This one seems at least unlikely to cause issues.

**Comment by kennylevinsen** [comment]
Yeah, keyword values seem to be pretty garbage (e.g. Evince has "ps" as keyword).

So, there's no field that can consistently be used to retrieve the GNOME product name. Great.

**Comment by pedrocr** [comment]
Hopefully this one doesn't have false positives at least.

File: src/widgets/launcher.rs (hunk 1 above)
Comment (kennylevinsen) [reply src/widgets/launcher.rs:245]: Thought: Change the iterator above to return `(score, gnome_score, desktop)` tuples, sort using the biggest one below, and then in draw, either show `name`, or `name (gnome_name)`, depending on whether the gnome name had a higher score.

Dunno if that'll look/feel better. Just a thought. Could also be used for keyword and/or alternate language matching.

**Comment by pedrocr** [comment]
In the latest gedit version `X-GNOME-Bugzilla-Product` is no longer present in the desktop file. Matching on keywords seems the only way forward but those need to be last priority as they contain a lot of stuff that can clash.

**Comment by pedrocr** [comment]
Closed in favor of #26

**Pull request closed without merging**